src/storage/v2/indices/vector_index.cpp

```diff
@@ -126,108 +62,107 @@ static const std::unordered_map<unum::usearch::metric_kind_t, std::function<doub
+                              IndexItem{std::make_shared<utils::Synchronized<mg_vector_index_t, std::shared_mutex>>(
+                                            std::move(mg_vector_index.index)),
+                                        spec});
+
+    // Update the index with the vertices
+    for (auto &vertex : vertices) {
+      UpdateVectorIndex(&vertex, LabelPropKey{spec.label, spec.property});
     }
   } catch (const std::exception &e) {
-    throw std::invalid_argument("Error parsing vector index spec: " + std::string(e.what()));
+    spdlog::error("Failed to create vector index {}: {}", spec.index_name, e.what());
+    return false;
   }
-
-  return result;
+  return true;
 }
 
-void VectorIndex::CreateIndex(const VectorIndexSpec &spec) {
-  const unum::usearch::metric_kind_t metric_kind = GetMetricKindFromConfig(spec.metric);
-  const unum::usearch::scalar_kind_t scalar_kind = GetScalarKindFromConfig(spec.scalar);
-
-  const unum::usearch::metric_punned_t metric(spec.dimension, metric_kind, scalar_kind);
-
-  // use the number of workers as the number of possible concurrent index operations
-  const unum::usearch::index_limits_t limits(spec.size_limit, FLAGS_bolt_num_workers);
-
-  const auto label_prop = LabelPropKey{spec.label, spec.property};
-  pimpl->index_name_to_label_prop_.emplace(spec.index_name, label_prop);
-  pimpl->index_.emplace(label_prop, IndexItem{mg_vector_index_t::make(metric), spec, utils::RWSpinLock()});
-  if (pimpl->index_[label_prop].mg_index.try_reserve(limits)) {
-    spdlog::info("Created vector index " + spec.index_name);
-  } else {
-    throw std::invalid_argument("Failed to create vector index " + spec.index_name +
-                                " due to failed memory allocation. Try again with a smaller size limit.");
+bool VectorIndex::DropIndex(std::string_view index_name) {
+  auto it = pimpl->index_name_to_label_prop_.find(index_name.data());
+  if (it == pimpl->index_name_to_label_prop_.end()) {
+    return false;
   }
+  const auto &label_prop = it->second;
+  pimpl->index_.erase(label_prop);
+  pimpl->index_name_to_label_prop_.erase(it);
+  spdlog::info("Dropped vector index {}", index_name);
+  return true;
 }
 
-void VectorIndex::UpdateVectorIndex(Vertex *vertex, const LabelPropKey &label_prop, const PropertyValue *value) const {
-  auto &[index, spec, lock] = pimpl->index_.at(label_prop);
+void VectorIndex::Clear() {
+  pimpl->index_name_to_label_prop_.clear();
+  pimpl->index_.clear();
+}
 
-  // first, try to remove entry (if it exists) and then add new one
+void VectorIndex::UpdateVectorIndex(Vertex *vertex, const LabelPropKey &label_prop, const PropertyValue *value) {
+  auto &[mg_index, spec] = pimpl->index_.at(label_prop);
+  bool is_index_full = false;
+  // try to remove entry (if it exists) and then add a new one + check if index is full
   {
-    auto guard = std::shared_lock{lock};
-    index.remove(vertex);
+    auto locked_index = mg_index->NonConstReadLock();
```

Comment:
Put a comment here to explain that for remove, even though it modifies, it is a concurrently safe operation. Hence MutableSharedLock is ok.

Also put documentation around MutableSharedLock that it is only safe to use for objects which are have concurrently safe modify operations.